src/particles.cpp

```diff
@@ -251,6 +252,16 @@ void ParticleSpawner::step(float dtime, ClientEnvironment* env)
 {
 	m_time += dtime;
 
+	bool unloaded = false;
+	v3f attached_offset = v3f(0,0,0);
+	if (m_attached_id != 0) {
+		ClientActiveObject *attached = env->getActiveObject(m_attached_id);
+		if (attached)
```

Comment:
no because it's used in the condition, in fact with this method you do the test for the decl/assignment directly without creating a variable out of the needed scope